image-hotspot-question.js

```diff
@@ -338,6 +347,58 @@ H5P.ImageHotspotQuestion = (function ($, Question) {
     return this.maxScore;
   };
 
+  /**
+   * Trigger xAPI answered event
+   */
+  ImageHotspotQuestion.prototype.triggerAnswered = function () {
+    var self = this;
+    var xAPIEvent = self.createXAPIEventTemplate('answered');
+    self.addQuestionToXAPI(xAPIEvent);
+    self.trigger(xAPIEvent);
+  };
+
+  /**
+   * Get xAPI data.
+   * Contract used by report rendering engine.
+   * 
+   * @see contract at {@link https://h5p.org/documentation/developers/contracts#guides-header-6}
+   */
+  ImageHotspotQuestion.prototype.getXAPIData = function () {
+    var self = this;
+    var xAPIEvent = self.createXAPIEventTemplate('answered');
+    xAPIEvent.setScoredResult(self.getScore(), self.getMaxScore(), self, true, true);
+    self.addQuestionToXAPI(xAPIEvent);
+    return {
+      statement: xAPIEvent.data.statement
+    };
+  };
+
+  /**
+   * Add the question itselt to the definition part of an xAPIEvent
+   */
+  ImageHotspotQuestion.prototype.addQuestionToXAPI = function (xAPIEvent) {
+    var definition = xAPIEvent.getVerifiedStatementValue(['object', 'definition']);
+    $.extend(true, definition, this.getxAPIDefinition());
+  };
+
+  /**
+   * Generate xAPI object definition used in xAPI statements.
+   * @return {Object}
+   */
+  ImageHotspotQuestion.prototype.getxAPIDefinition = function () {
+    // Individual report not supported
+    if (this.isRoot()) {
+      return;
+    }
+    var definition = {};
+    definition.description = {
+      'en-US': this.getTitle()
+    };
+    definition.type = 'http://adlnet.gov/expapi/activities/cmi.interaction';
+    definition.interactionType = 'general';
```

Comment:
I don't think that's a valid value for `interactionType`: https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#interaction-activities. What about `other`?

Reply:
Thanks, I will take care in next commit.